Extension/src/LanguageServer/settings.ts

```diff
@@ -903,3 +1035,158 @@ export function getEditorConfigSettings(fsPath: string): Promise<any> {
+        vcFormatSpaceWithinInitializerListBraces: settings.vcFormatSpaceWithinInitializerListBraces,
+        vcFormatSpacePreserveInInitializerList: settings.vcFormatSpacePreserveInInitializerList,
+        vcFormatSpaceBeforeOpenSquareBracket: settings.vcFormatSpaceBeforeOpenSquareBracket,
+        vcFormatSpaceWithinSquareBrackets: settings.vcFormatSpaceWithinSquareBrackets,
+        vcFormatSpaceBeforeEmptySquareBrackets: settings.vcFormatSpaceBeforeEmptySquareBrackets,
+        vcFormatSpaceBetweenEmptySquareBrackets: settings.vcFormatSpaceBetweenEmptySquareBrackets,
+        vcFormatSpaceGroupSquareBrackets: settings.vcFormatSpaceGroupSquareBrackets,
+        vcFormatSpaceWithinLambdaBrackets: settings.vcFormatSpaceWithinLambdaBrackets,
+        vcFormatSpaceBetweenEmptyLambdaBrackets: settings.vcFormatSpaceBetweenEmptyLambdaBrackets,
+        vcFormatSpaceBeforeComma: settings.vcFormatSpaceBeforeComma,
+        vcFormatSpaceAfterComma: settings.vcFormatSpaceAfterComma,
+        vcFormatSpaceRemoveAroundMemberOperators: settings.vcFormatSpaceRemoveAroundMemberOperators,
+        vcFormatSpaceBeforeInheritanceColon: settings.vcFormatSpaceBeforeInheritanceColon,
+        vcFormatSpaceBeforeConstructorColon: settings.vcFormatSpaceBeforeConstructorColon,
+        vcFormatSpaceRemoveBeforeSemicolon: settings.vcFormatSpaceRemoveBeforeSemicolon,
+        vcFormatSpaceInsertAfterSemicolon: settings.vcFormatSpaceInsertAfterSemicolon,
+        vcFormatSpaceRemoveAroundUnaryOperator: settings.vcFormatSpaceRemoveAroundUnaryOperator,
+        vcFormatSpaceAroundBinaryOperator: settings.vcFormatSpaceAroundBinaryOperator,
+        vcFormatSpaceAroundAssignmentOperator: settings.vcFormatSpaceAroundAssignmentOperator,
+        vcFormatSpacePointerReferenceAlignment: settings.vcFormatSpacePointerReferenceAlignment,
+        vcFormatSpaceAroundTernaryOperator: settings.vcFormatSpaceAroundTernaryOperator,
+        vcFormatWrapPreserveBlocks: settings.vcFormatWrapPreserveBlocks,
+        doxygenGenerateOnType: settings.doxygenGenerateOnType,
+        doxygenGeneratedStyle: settings.doxygenGeneratedCommentStyle,
+        filesExclude: otherSettings.filesExclude,
+        filesAutoSaveAfterDelay: otherSettings.filesAutoSaveAfterDelay,
+        filesEncoding: otherSettings.filesEncoding,
+        searchExclude: otherSettings.searchExclude,
+        editorAutoClosingBrackets: otherSettings.editorAutoClosingBrackets,
+        editorInlayHintsEnabled: otherSettings.InlayHintsEnabled
+    };
+};
+
+export function getAllWorkspaceFolderSettings(): WorkspaceFolderSettingsParams[] {
+    const workspaceFolderSettingsParams: WorkspaceFolderSettingsParams[] = [];
+    if (vscode.workspace.workspaceFolders && vscode.workspace.workspaceFolders.length > 0) {
+        for (const workspaceFolder of vscode.workspace.workspaceFolders) {
+            const client: Client = clients.getClientFor(workspaceFolder.uri);
```

Comment:
clients is undefined at this point, because it's invoked during the call to `    clients = new ClientCollection();`.